core/src/test/java/org/apache/iceberg/io/TestAppenderFactory.java

```diff
@@ -0,0 +1,265 @@
+    DataFile dataFile = prepareDataFile(rowSet, appenderFactory, outputFileFactory);
+
+    table.newRowDelta()
+        .addRows(dataFile)
+        .commit();
+
+    Assert.assertEquals("Should have the expected records.", expectedRowSet(rowSet), actualRowSet("*"));
+  }
+
+  @Test
+  public void testEqDeleteWriter() throws IOException {
+    List<Integer> equalityFieldIds = Lists.newArrayList(table.schema().findField("id").fieldId());
+    FileAppenderFactory<T> appenderFactory = createAppenderFactory(equalityFieldIds,
+        table.schema().select("id"), null);
+    OutputFileFactory outputFileFactory = createFileFactory();
+
+    List<T> rowSet = testRowSet();
+    DataFile dataFile = prepareDataFile(rowSet, appenderFactory, outputFileFactory);
+
+    table.newRowDelta()
+        .addRows(dataFile)
+        .commit();
+
+    List<T> deletes = Lists.newArrayList(
+        createRow(1, "aaa"),
+        createRow(3, "bbb"),
+        createRow(5, "ccc")
+    );
+    EqualityDeleteWriter<T> eqDeleteWriter =
+        appenderFactory.newEqDeleteWriter(outputFileFactory.newOutputFile(), format, partition);
+    try (EqualityDeleteWriter<T> closeableWriter = eqDeleteWriter) {
+      closeableWriter.deleteAll(deletes);
+    }
+
+    table.newRowDelta()
+        .addDeletes(eqDeleteWriter.toDeleteFile())
+        .commit();
+
+    List<T> expected = Lists.newArrayList(
+        createRow(2, "bbb"),
+        createRow(4, "ddd")
+    );
+    Assert.assertEquals("Should have the expected records", expectedRowSet(expected), actualRowSet("*"));
+  }
+
+  @Test
+  public void testPosDeleteWriter() throws IOException {
+    // Initialize FileAppenderFactory without pos-delete row schema.
+    FileAppenderFactory<T> appenderFactory = createAppenderFactory(null, null, null);
+    OutputFileFactory outputFileFactory = createFileFactory();
+
+    List<T> rowSet = testRowSet();
+    DataFile dataFile = prepareDataFile(rowSet, appenderFactory, outputFileFactory);
+
+    List<Pair<CharSequence, Long>> deletes = Lists.newArrayList(
+        Pair.of(dataFile.path(), 0L),
+        Pair.of(dataFile.path(), 2L),
+        Pair.of(dataFile.path(), 4L)
+    );
+
+    PositionDeleteWriter<T> eqDeleteWriter =
+        appenderFactory.newPosDeleteWriter(outputFileFactory.newOutputFile(), format, partition);
+    try (PositionDeleteWriter<T> closeableWriter = eqDeleteWriter) {
+      for (Pair<CharSequence, Long> delete : deletes) {
+        closeableWriter.delete(delete.first(), delete.second());
+      }
+    }
+
+    table.newRowDelta()
+        .addRows(dataFile)
+        .addDeletes(eqDeleteWriter.toDeleteFile())
+        .validateDataFilesExist(eqDeleteWriter.referencedDataFiles())
+        .validateDeletedFiles()
+        .commit();
+
+    List<T> expected = Lists.newArrayList(
+        createRow(2, "bbb"),
+        createRow(4, "ddd")
+    );
+    Assert.assertEquals("Should have the expected records", expectedRowSet(expected), actualRowSet("*"));
```

Comment:
Similar to above, I think this should check that only the path and position columns are written to the file and that they are the expected values. The test below should check that the `row` column is present and set correctly for each row.